### PR TITLE
perf: speed up `DoubleCentralizer.coeHom`

### PR DESCRIPTION
**Pull request by j-loreaux**
This declaration was *extremely slow*, and these changes speed it up significantly.

---

Went from >30 seconds to <5 seconds on my machine.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/Star/Multiplier.lean [hunk 1]
```diff
@@ -424,12 +424,9 @@ variable [StarRing 𝕜] [StarRing A] [StarModule 𝕜 A] [NormedStarGroup A]
...
   star a :=
-    { fst :=
-        (((starₗᵢ 𝕜 : A ≃ₗᵢ⋆[𝕜] A) : A →L⋆[𝕜] A).comp a.snd).comp
-          ((starₗᵢ 𝕜 : A ≃ₗᵢ⋆[𝕜] A) : A →L⋆[𝕜] A)
-      snd :=
-        (((starₗᵢ 𝕜 : A ≃ₗᵢ⋆[𝕜] A) : A →L⋆[𝕜] A).comp a.fst).comp
-          ((starₗᵢ 𝕜 : A ≃ₗᵢ⋆[𝕜] A) : A →L⋆[𝕜] A)
+    { toProd :=
+        (((starₗᵢ 𝕜 (E := A) : A →L⋆[𝕜] A).comp a.snd).comp (starₗᵢ 𝕜 (E := A) : A →L⋆[𝕜] A),
+        ((starₗᵢ 𝕜 (E := A) : A →L⋆[𝕜] A).comp a.fst).comp (starₗᵢ 𝕜 (E := A) : A →L⋆[𝕜] A))
```
Comment (eric-wieser) [inline Mathlib/Analysis/NormedSpace/Star/Multiplier.lean:429]: I claim these are syntactically identical to Lean after desugaring; am I incorrect?

Comment (j-loreaux) [reply Mathlib/Analysis/NormedSpace/Star/Multiplier.lean:429]: no, you're right. I was mainly trying to stick to the `toProd` paradigm to match, but in this case it's no different.

Comment (j-loreaux) [reply Mathlib/Analysis/NormedSpace/Star/Multiplier.lean:429]: Sorry, I shouldn't have included that as part of this PR (or mentioned that it was just clean-up).

File: Mathlib/Analysis/NormedSpace/Star/Multiplier.lean [hunk 2]
```diff
@@ -475,12 +472,10 @@ maps `Lₐ Rₐ : A →L[𝕜] A` given by left- and right-multiplication by `a`
...
 `algebraMap 𝕜 𝓜(𝕜, 𝕜)` coercion, but these are propositionally equal. See
 `DoubleCentralizer.coe_eq_algebraMap` below. -/
--- porting note: added `noncomputable` because an IR check failed?
 @[coe]
-protected noncomputable def coe (a : A) : 𝓜(𝕜, A) :=
-  { fst := ContinuousLinearMap.mul 𝕜 A a
-    snd := (ContinuousLinearMap.mul 𝕜 A).flip a
-    central := fun _x _y => mul_assoc _ _ _ }
+protected def coe (a : A) : 𝓜(𝕜, A) where
+  toProd := (mul 𝕜 A a, (mul 𝕜 A).flip a)
```
Comment (eric-wieser) [inline Mathlib/Analysis/NormedSpace/Star/Multiplier.lean:477]: This one also should have no effect

**Comment by eric-wieser** [comment]
is this faster or slower?
```
noncomputable def coeHom [StarRing 𝕜] [StarRing A] [StarModule 𝕜 A] [NormedStarGroup A] :
    A →⋆ₙₐ[𝕜] 𝓜(𝕜, A) where
  toFun a := a
  map_smul' _ _ := ext _ _ _ _ <| Prod.ext (map_smul _ _ _) (map_smul _ _ _)
  map_zero' := ext _ _ _ _ <| Prod.ext (map_zero _) (map_zero _)
  map_add' _ _ := ext _ _ _ _ <| Prod.ext (map_add _ _ _) (map_add _ _ _)
  map_mul' _ _ :=  ext _ _ _ _ <| Prod.ext
    (ContinuousLinearMap.ext fun _ => (mul_assoc _ _ _))
    (ContinuousLinearMap.ext fun _ => (mul_assoc _ _ _).symm)
  map_star' _ := ext _ _ _ _ <| Prod.ext
    (ContinuousLinearMap.ext fun _ => (star_star_mul _ _).symm)
    (ContinuousLinearMap.ext fun _ => (star_mul_star _ _).symm)
 ```

**Comment by j-loreaux** [comment]
closing in favor of #6329

**Pull request closed without merging**